### PR TITLE
[UR][L0] Fix kernel error handling and enumeration checking

### PR DESCRIPTION
**Pull request by nrspruit**
pre-commit PR for https://github.com/oneapi-src/unified-runtime/pull/1810

### CODE REVIEW

**Comment by omarahmed1111** [comment]
@nrspruit Could you please update this so the UR PR is included in the next UR bump? Thanks!

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 1]
```diff
@@ -115,23 +115,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-
-  set(UMF_BUILD_EXAMPLES OFF CACHE INTERNAL "EXAMPLES")
-  # Due to the use of dependentloadflag and no installer for UMF and hwloc we need
-  # to link statically on windows
-  if(WIN32)
-    set(UMF_BUILD_SHARED_LIBRARY OFF CACHE INTERNAL "Build UMF shared library")
-    set(UMF_LINK_HWLOC_STATICALLY ON CACHE INTERNAL "static HWLOC")
-  else()
-    set(UMF_DISABLE_HWLOC ${SYCL_UMF_DISABLE_HWLOC} CACHE INTERNAL "Disable hwloc for UMF")
-  endif()
```
Comment (omarahmed1111) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:134]: I think these changes are the ones making the CI failure, Is this related to the PR changes?

Comment (omarahmed1111) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:134]: gentle ping! @nrspruit as if this changes are not needed we could remove them and rerun CI.

Comment (nrspruit) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake]: Fixed in my latest rebase.

**Comment by omarahmed1111** [comment]
This was covered as part of this [PR](https://github.com/intel/llvm/pull/14944)

**Pull request closed without merging**